src/raw_to_processed.py

```diff
@@ -0,0 +1,222 @@
+import os
+import pandas as pd
+from glob import glob
+from helpers import load_cfg
+from minio import Minio
+
+###############################################
+# Parameters & Arguments
+###############################################
+DATA_PATH = "data/"
+YEARS = ["2022", "2023"]
+TAXI_LOOKUP_PATH = "src/data/taxi_lookup.csv"
+CFG_FILE =  "config/datalake.yaml"
+###############################################
+
+
+###############################################
+# Utils
+###############################################
+def create_bucket(bucket_name):
+     # Load minio config
+    cfg = load_cfg(CFG_FILE)
+    datalake_cfg = cfg["datalake"]
+
+    # Create a client with the MinIO server
+    minio_client = Minio(
+        endpoint=datalake_cfg["endpoint"],
+        access_key=datalake_cfg["access_key"],
+        secret_key=datalake_cfg["secret_key"],
+        secure=False,
+    )
+
+    # Create bucket if not exist
+    found = minio_client.bucket_exists(bucket_name=bucket_name)
+    if not found:
+        minio_client.make_bucket(bucket_name=bucket_name)
+    else:
+        print(f"Bucket {bucket_name} already exists, skip creating!")
+###############################################
+
+
+###############################################
+# Process data
+###############################################
+def drop_column(df, file):
+    """
+        Drop columns 'store_and_fwd_flag'
+    """
+    if "store_and_fwd_flag" in df.columns:
+        df = df.drop(columns=["store_and_fwd_flag"])
+        print("Dropped column store_and_fwd_flag from file: " + file)
+    else:
+        print("Column store_and_fwd_flag not found in file: " + file)
+
+    return df
+
+
+def merge_taxi_zone(df, file):
+    """
+        Merge dataset with taxi zone lookup
+    """
+    df_lookup = pd.read_csv(TAXI_LOOKUP_PATH)
+
+    if "pickup_latitude" not in df.columns:
+        # merge for pickup locations
+        df = df.merge(df_lookup, left_on="pulocationid", right_on="LocationID")
+        df = df.drop(columns=["LocationID", "Borough", "service_zone", "zone"])
+        df = df.rename(columns={
+            "latitude" : "pickup_latitude",
+            "longitude" : "pickup_longitude"
+        })
+    
+    if "dropoff_latitude" not in df.columns:
```

Comment:
This code snippet is fairy similar to the code `if "pickup_latitude" not in df.columns:`, try to create functions!

Reply:
Fixed!